src/Dotnet.Script.Core/ScriptEmitResult.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Microsoft.CodeAnalysis;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+
+namespace Dotnet.Script.Core
+{
+    public class ScriptEmitResult
+    {
+        private ScriptEmitResult() { }
+
+        public ScriptEmitResult(MemoryStream peStream, MemoryStream pdbStream, IEnumerable<MetadataReference> directiveReferences)
+        {
+            PeStream = peStream;
+            PdbStream = pdbStream;
+            DirectiveReferences = directiveReferences.ToImmutableArray();
+        }
+
+        public MemoryStream PeStream { get; }
+        public MemoryStream PdbStream { get; }
+        public ImmutableArray<Diagnostic> Diagnostics { get; private set; } = ImmutableArray.Create<Diagnostic>();
+        public ImmutableArray<MetadataReference> DirectiveReferences { get; } = ImmutableArray.Create<MetadataReference>();
+        public bool IsErrored => Diagnostics.Any();
```

Comment:
`Success` might be a better name her since that aligns with `EmitResult.Success`